src/osdc/Objecter.cc

```diff
@@ -3435,6 +3435,7 @@ void Objecter::handle_osd_op_reply(MOSDOpReply *m)
     // new tid
     s->ops.erase(op->tid);
     op->tid = ++last_tid;
+    s->ops[op->tid] = op; // add op to ops again
```

Comment:
Oh, we probably want the localized reads to hit the above block too.

Actually I'm not sure we ever use EAGAIN where we don't want to. The only place I'm aware of us returning EAGAIN is when the replica notices the object is in recovery or other indeterminate state, but doesn't have the machinery to block the op or know when it can be safely processed. (The primary does.). But there may be other places with caching or something.